library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types/Shared.hs

```diff
@@ -36,6 +37,10 @@ type VariantToType = Map.T Symbol SumInfo
 -- numbering to go along with a given name.
 type Names = Map.T Symbol Int
 
+-- | A mapping of record names to LLVM types and lists of (name, type) pairs
+-- | (one for each field).
+type RecordTable = Map.T Symbol (Type, [(Symbol, Type)])
```

Comment:
This has a nice proper type mapping,

Though, for overlap we may eventually want
`[(Type, [(Symbol, Type)])]`

if there is an overlap for any reason, then we can pick the right one if there is a type annotation, and make them match. 

Though maybe we should find a way to disambiguate by core?

Reply:
I think what I'll at least try first here is to make the record types truly scoped as I described in https://github.com/anoma/juvix/pull/1106#issuecomment-1013715909 , thus giving Core the chance to use that mechanism to handle declarations in different modules and/or declarations whose names shadow those of other record declarations.  Then the way of having multiple record declarations with the same symbol would be that the `recordTab` would be saved and restored around the term wrapped by a `ScopedRecordDecl`.